### DISCUSSION TITLE
Join tables with foreign keys automatiically [Apr 23]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#1939] [Discord Threads]
> 🙋‍♂️ **BoMBx**

**Comment by o1lab** [comment]
> 👨‍🚀 **wingkwong**  
#### Yes and No. Columns with LinkToAnotherRecord type will be created.  
#  
> 👨‍🚀 **wingkwong**  
#### Discussed with user in DM. Here's the summary.   
####    
#### Question 1: Want to query a table in REST API call but excluding some fields   
####    
#### Ans: Using fields to determine which fields to show.    
####    
#### Example:   
#### /api/v1/db/data/{orgs}/{projectName}/{tableName}/views/{viewName}?fields= field_name,field_name2   
####    
#### Question 2: Query a table joining another table with some identical column names. Would it be any conflicts.    
####    
#### Answer:    
####    
#### In the returned json, the LinkToAnotherRecord data is presented in a nested way. For example, table A and table B both have a ManyToMany column.   
#### In GUI, you will see TableBMMList in TableA and TableAMMList in Table B   
#### if you call the api, you should see    
####    
#### {   
####   field1: 'xxx',   
####   field2: 'xxx',   
####   TableAMMList: [{...}, {...}, {...}],   
####   created_by: 'xxx'   
#### }   
####    
#### Similarly, for HasMany column, it will be TableARead   
####    
#### {   
####   field1: 'xxx',   
####   field2: 'xxx',   
####   TableARead: {...},   
####   created_by: 'xxx'   
#### }   
####    
#### so one can identify TableARead.field1 and field1  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/966928425665323008)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
